experimental/PropertyDDS/packages/property-properties/src/properties/baseProperty.js

```diff
@@ -72,1227 +73,1224 @@ var PATH_TOKENS = {
+            typeid: 'BaseProperty',
+            isConstant: false
+        };
+
+        in_params = _.merge(defaultParams, in_params);
+
+        if (this._id !== in_params.id) {
+            this._id = in_params.id;
+        }
 
-/**
- * Default constructor for BaseProperty
- * @param {object} in_params List of parameters
- * @param {string} in_params.id id of the property
- * @param {string=} [in_params.typeid='BaseProperty'] The type unique identifier
- * @param {number=} [in_params.length=1] The length of the property. Only valid if
- *   the property is an array, otherwise the length defaults to 1
- * @param {string=} [in_params.context='single'] The type of property this template represents
- *   i.e. single, array, map, set.
- * @param {Array.<object>} in_params.properties List of property templates that
- *   are used to define children properties -- UNUSED PARAMETER ??
- * @param {Array.<string>} in_params.inherits List of property template typeids that this
- *   PropertyTemplate inherits from -- UNUSED PARAMETER ??
- * @abstract
- * @constructor
- * @protected
- * @alias property-properties.BaseProperty
- * @category Properties
- */
-var BaseProperty = function (in_params) {
-    // Pre-conditions
-    ConsoleUtils.assert(in_params, MSG.PROP_CONSTRUCTOR_EXPECTS_OBJECTS);
-
-    in_params = in_params || {};
-
-    if (this._id !== in_params.id) {
-        this._id = in_params.id;
-    }
-
-    // Makes sure context value is fine
-    ConsoleUtils.assert(!in_params.context || in_params.context === this._context,
-        MSG.CONTEXT_NOT_AS_EXPECTED + this._context + ' != ' + in_params.context);
-
-    // Sets typeid if default value is not fine
-    if (in_params.typeid && in_params.typeid !== this._typeid) {
-        this._typeid = in_params.typeid;
-    }
-
-    this._parent = undefined;
+        // Makes sure context value is fine
+        ConsoleUtils.assert(!in_params.context || in_params.context === this._context,
+            MSG.CONTEXT_NOT_AS_EXPECTED + this._context + ' != ' + in_params.context);
 
-    // internal management
-    if (!this._noDirtyInBase) {
-        this._dirty = MODIFIED_STATE_FLAGS.CLEAN;
-    }
-};
+        // Sets typeid if default value is not fine
+        if (in_params.typeid && in_params.typeid !== this._typeid) {
+            this._typeid = in_params.typeid;
+        }
 
-BaseProperty.MODIFIED_STATE_FLAGS = MODIFIED_STATE_FLAGS;
-BaseProperty.prototype.MODIFIED_STATE_FLAGS = MODIFIED_STATE_FLAGS;
+        this._parent = undefined;
+        this._isConstant = in_params.isConstant;
 
-BaseProperty.REFERENCE_RESOLUTION = REFERENCE_RESOLUTION;
-BaseProperty.prototype.REFERENCE_RESOLUTION = REFERENCE_RESOLUTION;
+        // internal management
+        if (!this._noDirtyInBase) {
+            this._dirty = MODIFIED_STATE_FLAGS.CLEAN;
+        }
+    };
 
-BaseProperty.PATH_TOKENS = PATH_TOKENS;
-BaseProperty.prototype.PATH_TOKENS = PATH_TOKENS;
+    static MODIFIED_STATE_FLAGS = MODIFIED_STATE_FLAGS;
```

Comment:
Why the same property is exposed as static and in the prototype? Perhaps we should keep only one ?